generator-engine/src/main/java/hu/blackbelt/judo/meta/psm/generator/engine/PsmGenerator.java

```diff
@@ -333,4 +343,49 @@ public static void recalculateChecksumForDirectory(PsmGeneratorParameter paramet
         ModelGenerator.recalculateChecksumToDirectory(genericParams, applications);
     }
 
+    public static String generalizeName(String str) {
+        if (str == null || str.length() == 0) {
+            return str;
+        }
+        boolean isUpperSnakeCae = !str.matches(negateRegex("([A-Z_0-9]*)"));
```

Comment:
typo in var name and why isn't it `str.matches("[A-Z_0-9]+");` if the expression is negated twice?

Reply:
Because the negated regexp not the same with that negate the matches itself.